src/overlays/actors/ovl_Bg_Mizu_Water/z_bg_mizu_water.c

```diff
@@ -21,19 +43,305 @@ const ActorInit Bg_Mizu_Water_InitVars = {
     (ActorFunc)BgMizuWater_Update,
     (ActorFunc)BgMizuWater_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Water/func_8089F8B0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Water/func_8089F9D0.s")
+static u32 unk_808A0550 = 0;
+static f32 unk_808A0554 = 110.0f;
+
+static u32 waterBoxIDs[] = { 0x00000002, 0x00000003, 0x00000005, 0x00000007,
+                             0x0000000C, 0x00000014, 0x00000015, 0x00000016 };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F(scale, 1, ICHAIN_STOP),
+};
+
+u32 BgMizuWater_getWaterLevelActionIndex(s16 switchID, GlobalContext* globalCtx) {
+    u32 ret;
+
+    if (bREG(0) != 0) {
+        switch (bREG(1)) {
+            case 0:
+                Flags_SetSwitch(globalCtx, 0x1C);
+                break;
+            case 1:
+                Flags_SetSwitch(globalCtx, 0x1D);
+                break;
+            case 2:
+                Flags_SetSwitch(globalCtx, 0x1E);
+                break;
+        }
+        bREG(0) = 0;
+    }
+    if ((Flags_GetSwitch(globalCtx, 0x1C)) && (switchID != 0x1C))
+        ret = 3;
+    else if ((Flags_GetSwitch(globalCtx, 0x1D)) && (switchID != 0x1D))
+        ret = 2;
+    else if ((Flags_GetSwitch(globalCtx, 0x1E)) && (switchID != 0x1E))
+        ret = 1;
+    else
+        ret = 0;
+
+    return ret;
```

Comment:
you have to use a temp for the return value?

Reply:
Yes, it is required.